app/src/main/java/net/gsantner/opoc/ui/SearchOrCustomTextDialog.java

```diff
@@ -38,26 +41,33 @@
 import android.view.Window;
 import android.view.WindowManager;
 import android.widget.ArrayAdapter;
+import android.widget.Button;
 import android.widget.Filter;
+import android.widget.ImageView;
 import android.widget.LinearLayout;
 import android.widget.ListView;
 import android.widget.TextView;
 
+import net.gsantner.markor.R;
 import net.gsantner.opoc.util.Callback;
 import net.gsantner.opoc.util.ContextUtils;
 
 import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
 import java.util.List;
 import java.util.Locale;
+import java.util.Set;
 import java.util.regex.Pattern;
 
 public class SearchOrCustomTextDialog {
 
     public static class DialogOptions {
-        public Callback.a1<String> callback;
-        public Callback.a2<String, Integer> withPositionCallback;
-        public List<? extends CharSequence> data;
-        public List<? extends CharSequence> highlightData;
+        public Callback.a1<String> callback = null;
+        public Callback.a1<Integer> positionCallback = null;
+        public Callback.a1<List<Integer>> multiSelectCallback = null;
```

Comment:
To be honest: I prefer to have a single callback ("way out" / single return value). Thats what I also thought when I stumbled across it recently with the second "positionCallback" addition.

My suggestion is:
* `Callback.a1<String, MapPropertyBackend<String>>`
* with the first parameter being the usual selection (i.e. first selection)
* second parameter is a hashmap with getString / getInt etc
* There you can store i.e. position int and any additional data not required for the simple search

Reply:
I thought of unifying all the callbacks into a single  `Pair<String, List<Integer>>`. The problem here is the calling code has to then have logic to decide what to do based on what flags were set and what is in the callback.

With the current system in this PR we have

1. `callback`: string in search or item selected
2. `positionCallback` : position of item selected
3. `multiSelectCallback`: all selected positions. Also serves as a flag that multi-select is enabled.

Calling code just sets the appropriate callback(s) to do the correct thing and doesn't have to have any if-then logic.

One option we do to simplify things here is unify the position callbacks and have an explicit multi-select flag.

Reply:
Only place we need one position:   In editor content search

Places where multiple positions:  
todo tag  
todo project  

everything else just needs the string result, correct?

Reply:
what do you think off `<String, Integer...>`? Should cover all cases as well